bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/findandreplace/FindReplaceDialog.java

```diff
@@ -62,29 +60,25 @@
 import org.eclipse.jface.text.FindReplaceDocumentAdapter;
 import org.eclipse.jface.text.FindReplaceDocumentAdapterContentProposalProvider;
 import org.eclipse.jface.text.IFindReplaceTarget;
-import org.eclipse.jface.text.IFindReplaceTargetExtension;
-import org.eclipse.jface.text.IFindReplaceTargetExtension3;
 import org.eclipse.jface.text.IFindReplaceTargetExtension4;
-import org.eclipse.jface.text.IRegion;
-import org.eclipse.jface.text.Region;
 import org.eclipse.jface.text.TextUtilities;
 
-import org.eclipse.ui.IEditorPart;
-import org.eclipse.ui.IWorkbenchPage;
-import org.eclipse.ui.IWorkbenchWindow;
 import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.fieldassist.ContentAssistCommandAdapter;
-import org.eclipse.ui.internal.texteditor.NLSUtility;
 import org.eclipse.ui.internal.texteditor.SWTUtil;
 
+import org.eclipse.ui.texteditor.IAbstractTextEditorHelpContextIds;
+import org.eclipse.ui.texteditor.ITextEditorActionDefinitionIds;
+
 
 /**
  * Find/Replace dialog. The dialog is opened on a particular
  * target but can be re-targeted. Internally used by the <code>FindReplaceAction</code>
  */
-class FindReplaceDialog extends Dialog {
+public class FindReplaceDialog extends Dialog {
```

Comment:
The build logs
```
 There are API errors:
[ERROR] src/org/eclipse/ui/findandreplace/FindReplaceDialog.java:77 Missing @since tag on org.eclipse.ui.findandreplace.FindReplaceDialog
src/org/eclipse/ui/findandreplace/FindReplaceLogicFindAllStatus.java:7 Invalid @since 1.0 tag
...
[ERROR] META-INF/MANIFEST.MF:5 The minor version should be incremented in version 3.17.200, since new APIs have been added since version 3.17.100
```
Please take that seriously. If you add new API that should be crafted very carefully. line 76 documents this class is (only) used internally. That contradicts an API. Also in the test cases there are String constants referring to the class name. Isn't it possible to use the class name at runtime instead?

Reply:
I have made findandreplace an internal package:

```
Export-Package: 
 org.eclipse.ui.contentassist,
 org.eclipse.ui.findandreplace;x-friends="org.eclipse.ui.workbench.texteditor.tests";x-internal:=true,
```